files/bin/private-dns-name

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o xtrace
+
+# Retrieves the PrivateDnsName from EC2 for this instance, waiting until
+# it is available if necessary (due to eventual consistency).
+
+function log {
+  echo >&2 "$(date '+%Y-%m-%dT%H:%M:%S%z')" "[private-dns-name]" "$@"
+}
+
+INSTANCE_ID=$(imds /latest/meta-data/instance-id)
+
+# the AWS CLI currently constructs the wrong endpoint URL on localzones (the availability zone group will be used instead of the parent region)
+# more info: https://github.com/aws/aws-cli/issues/7043
+REGION=$(imds /latest/meta-data/placement/region)
+
+# by default, wait for ~120 seconds (depending on jitter)
+PRIVATE_DNS_NAME_MAX_ATTEMPTS=${PRIVATE_DNS_NAME_MAX_ATTEMPTS:-20}
+PRIVATE_DNS_NAME_ATTEMPT_INTERVAL=${PRIVATE_DNS_NAME_ATTEMPT_INTERVAL:-6}
+PRIVATE_DNS_NAME_JITTER=${PRIVATE_DNS_NAME_JITTER:-1}
+
+log "will make up to ${PRIVATE_DNS_NAME_MAX_ATTEMPTS} attempt(s) every ${PRIVATE_DNS_NAME_ATTEMPT_INTERVAL} second(s) with ${PRIVATE_DNS_NAME_JITTER} second(s) of jitter"
+
+ATTEMPT=0
+while true; do
+  PRIVATE_DNS_NAME=$(aws ec2 describe-instances --region $REGION --instance-ids $INSTANCE_ID | jq -r '.Reservations[].Instances[].PrivateDnsName')
+  if [ ! "${PRIVATE_DNS_NAME}" = "" ] || [ ${ATTEMPT} -ge ${PRIVATE_DNS_NAME_MAX_ATTEMPTS} ]; then
+    break
+  fi
+  ATTEMPT=$((ATTEMPT + 1))
+  JITTER=$(seq "-${PRIVATE_DNS_NAME_JITTER}" "${PRIVATE_DNS_NAME_JITTER}" | shuf -n1)
+  DELAY=$((PRIVATE_DNS_NAME_ATTEMPT_INTERVAL + JITTER))
```

Comment:
Can we just wait for regular interval, in this way we would know how long we have waited for before giving up?

Reply:
I don't feel too strongly either way; but there is a case in which the describe-instances call is throttled, and the jitter will help in that case. If you think the debuggability of a consistent interval is more worthwhile, I'll remove this

Reply:
If we want to handle the throttling scenarios, then we should do back-off retry. This isn't a blocking comment from my end. I am approving this PR for now.

Reply:
That's fair. I'll just remove the jitter to keep things simple, and we can add exponential backoff later if we decide it's necessary.